### PR TITLE
moved all relevant files into a src folder and added a docker file

### PR DESCRIPTION
**Pull request by wibed**

### CODE REVIEW

**Comment by wibed** [comment]
AttributeError: 'HttpResponseRedirect' object has no attribute '_headers'

seems to be the cause on all tests.
could it be that the tests have not been updated?

**Comment by peppelinux** [comment]
Ho @wibed, to claims:

1. Please push to Dev branch, can you rebase this PR?
2. Unittest failed

And more, I'm looking to decouple uniauth-saml2 Django app from uniAuth project. In the future I'd configure in tu a latter a future release of django-oidc-op. Does It sounds good to you?

uniAuth-saml2-idp would be installed as a puoi package

**Comment by peppelinux** [comment]
yes tests are broken, I'm moving to gh actions and I'll update the entire project

idp and uniauth app will be merged in a single app. I'll take some days before come back on this PR that probably shall be refactored

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+	libldap2-dev \
+	gcc
+
+
+FROM builder as virtenv
+
+RUN mkdir /app
+WORKDIR /app
+
+RUN pip install \
```
Comment (peppelinux) [inline Dockerfile:38]: why don't you use `pip install -r requirements.txt` here?

**Comment by peppelinux** [comment]
I've just fixed tests and moved relevant project files in the example folder.
more work has to be done from now.

ldap tests have been disabled in gh actions due to a very slow behaviour, with these coverage is up to 81%.
Feel free to adapt your PR on this latter update.

**Review by peppelinux — CHANGES_REQUESTED** [review]

**Comment by peppelinux** [comment]
Here:
https://github.com/UniversitaDellaCalabria/uniAuth/commit/7350cfaaf23b5a65cdda9ab9d1edac2447babcf1

any further updates here:
https://github.com/UniversitaDellaCalabria/uniAuth/issues/18

File: Dockerfile (hunk 1 above)
Comment (wibed) [reply Dockerfile:38]: it wasn't clear that requirements.txt and requirements-dev.txt are dependent on each other.
as the first should be the minimal one and the second an extensive one.

Comment (peppelinux) [reply Dockerfile:38]: requirements.txt is the minimal

-dev is for developers only
-docs for built sphinx
-customization for use additional backends, such ldap and other things.

For a good asset use the first and -customizations

Comment (wibed) [reply Dockerfile:38]: I use requirements-dev.txt at #54
sadly until I got it to work I had to install a multitude of dependencies. 

I installed them globally as it is better documented
and could moved to a local installation later

**Comment by wibed** [comment]
> Ho @wibed, two claims:
> 
> 1. Please push to Dev branch, can you rebase this PR?
> 2. Unittest failed
> 
> And more, I'm looking to decouple uniauth-saml2 Django app from uniAuth project. In the future I'd configure in this project a release of django-oidc-op. Does It sounds good to you?
> 
> uniAuth-saml2-idp would be installed as a pypi package

why don't you tag your releases and add push new features directly to master
don't get me wrong, I try to understand your decision

**Comment by wibed** [comment]
> Ho @wibed, two claims:
> 
> 1. Please push to Dev branch, can you rebase this PR?
> 2. Unittest failed
> 
> And more, I'm looking to decouple uniauth-saml2 Django app from uniAuth project. In the future I'd configure in this project a release of django-oidc-op. Does It sounds good to you?
> 
> uniAuth-saml2-idp would be installed as a pypi package

why don't you tag your releases and add push new features directly to master
don't get me wrong, I try to understand your decision

**Comment by peppelinux** [comment]
Actually master and Dev branch are sync.
Requirements files have been improved

**Comment by peppelinux** [comment]
please rebase and test if you can, I made a lot of breaking changes

**Comment by wibed** [comment]
please continue on #36

**Pull request closed without merging**